app/serializers/serialize_exercise_instructions.rb

```diff
@@ -0,0 +1,72 @@
+class SerializeExerciseInstructions
+  include Mandate
+
+  initialize_with :exercise
+
+  def call
+    {
+      overview: overview,
+      general_hints: general_hints,
+      tasks: tasks
+    }
+  end
+
+  private
+  def overview
+    instructions_doc.each.
+      take_while { |node| node.type != :header }.
+      map(&:to_html).
+      join
+  end
+
+  def general_hints
+    hints["general"].to_a
+  end
+
+  memoize
+  def hints
+    hints_doc.each_cons(2).each_with_object({}) do |(header, list), hints|
+      next unless header.type == :header
+      next unless header.header_level == 2
+      next unless list.type == :list
+
+      heading = parse_title(header).downcase
+      hints[heading] = list.each.map { |list_item| list_item.each.first.to_html }
+    end
+  end
+
+  def tasks
+    instructions_doc.each.
+      drop_while { |node| node.type != :header }.
+      chunk_while { |_, nxt| nxt.type != :header }.
+      each_with_object([]) do |nodes, tasks|
+        task_title = parse_title(nodes.first)
+        task_text = nodes.drop(1).each.map(&:to_html).join.strip
```

Comment:
This has a small hidden bug, as just concatenating the HTML results in spacing being lost. I have to check how the HTML would be rendered when the elements are combined.

Reply:
This has been fixed